src/ip/QosConfig.h

```diff
@@ -71,13 +80,14 @@ namespace Qos
 void getTosFromServer(const Comm::ConnectionPointer &server, fde *clientFde);
 
 /**
-* Function to retrieve the netfilter mark value of the connection
-* to the upstream server. Called by FwdState::dispatch if QOS
-* options are enabled.
-* @param server    Server side descriptor of connection to get mark for
-* @param clientFde Pointer to client side fde instance to set nfmarkFromServer in
+* Function to retrieve the netfilter mark value of the connection.
+* Called by FwdState::dispatch if QOS options are enabled or by
+* Comm::TcpAcceptor::acceptOne
+*
+* @param conn      Pointer to connection to get mark for
+* @param conn_type Specifies connection type (incoming or outgoing)
 */
-void getNfmarkFromServer(const Comm::ConnectionPointer &server, const fde *clientFde);
+nfmark_t getNfmarkFromConnection(const Comm::ConnectionPointer &conn, enum ConnType conn_type = Outgoing);
```

Comment:
Please remove the default parameter value. When there is no reasonable default, it is a good idea to make the caller think what it actually wants.